### PR TITLE
Mining disaster example tidy up

### PR DESCRIPTION
**Pull request by statwonk**
#### Submission Checklist

- [ ] Run unit tests: `./runTests.py src/test/unit`

```
Traceback (most recent call last):
  File "./runTests.py", line 16, in <module>
    "lib", "stan_math", "runTests.py"))
IOError: [Errno 2] No such file or directory
```

- [ ] Run cpplint: `make cpplint`

```
python: can't open file '/cpplint.py': [Errno 2] No such file or directory
make: *** [cpplint] Error 2
```

- [ x] Declare copyright holder and open-source license: see below

#### Summary

A simple pass tidying up an example.

#### Intended Effect

To reduce confusion among newcomers.

#### How to Verify

Read.

#### Side Effects

A larger community.

#### Copyright and Licensing

Please list the copyright holder for the work you are submitting (this will be you or your assignee, such as a university or company):

[Zapier](https://zapier.com)

By submitting this pull request, the copyright holder is agreeing to license the submitted work under the following licenses:
- Code: BSD 3-clause (https://opensource.org/licenses/BSD-3-Clause)
- Documentation: CC-BY 4.0 (https://creativecommons.org/licenses/by/4.0/)

### CODE REVIEW

**Comment by bob-carpenter** [comment]
`runTests.py`  is in the home directory of `stan-dev/stan`, and you need to run it from there.

`cpplint.py` is in the math lib.  That means when you're in the Stan repo, you need to check that out as a submodule.  You can do that using `make math-update` to get the synchronized version.

Anyway, not needed here.  I'm going to approve as is.  thanks.

File: src/docs/stan-reference/programs/mining-disasters/changepoint.stan [hunk 1]
```diff
@@ -46,28 +46,28 @@ data {
 }
 transformed data {
   real log_unif;
```
Comment (bob-carpenter) [inline src/docs/stan-reference/programs/mining-disasters/changepoint.stan]: If you want this to be even cleaner, it should be

```
real log_unif = -log(T);
```

File: src/docs/stan-reference/programs/mining-disasters/changepoint.stan [hunk 2]
```diff
@@ -46,28 +46,28 @@ data {
...
+  log_unif = -log(T);  // log p(s)
 }
 parameters {
   real<lower=0> e;    // before change disaster rate
   real<lower=0> l;    // after change disaster rate
 }
 transformed parameters {
   vector[T] lp;
-  lp <- rep_vector(log_unif, T);
+  lp = rep_vector(log_unif, T);
```
Comment (bob-carpenter) [inline src/docs/stan-reference/programs/mining-disasters/changepoint.stan]: same thing here with compound declare-define.

File: src/docs/stan-reference/programs/mining-disasters/changepoint.stan [hunk 3]
```diff
@@ -46,28 +46,28 @@ data {
...
   for (s in 1:T)
     for (t in 1:T)
-      lp[s] <- lp[s] + poisson_log(D[t], if_else(t < s, e, l));
+      lp[s] = lp[s] + poisson_lpmf(D[t] | t < s ? e : l);
```
Comment (bob-carpenter) [inline src/docs/stan-reference/programs/mining-disasters/changepoint.stan]: This should get cleaned up to the `+=` syntax:

```
lp[s] += poisson_lpmf(D[t] | t < s ? e : l);
```

File: src/docs/stan-reference/programs/mining-disasters/changepoint.stan [hunk 4]
```diff
@@ -46,28 +46,28 @@ data {
...
 
   // likelihood
-  increment_log_prob(log_sum_exp(lp));
-}    
+  target += log_sum_exp(lp);
+}
 generated quantities {
   int<lower=1,upper=T> s;
-  s <- categorical_rng(softmax(lp));
+  s = categorical_rng(softmax(lp));
```
Comment (bob-carpenter) [inline src/docs/stan-reference/programs/mining-disasters/changepoint.stan]: compound declare-define

**Comment by bob-carpenter** [comment]
Let me know if you'd like me to go ahead and merge or whether you want to keep improving.

**Comment by statwonk** [comment]
I'd like to make the improvements you suggested. I'm follow up next week and 'at' you when I'm finished. Thank you for the feedback!

**Comment by statwonk** [comment]
Thanks for the helpful hints @bob-carpenter. 👍

**Pull request merged**